### ISSUE TITLE
Is FFMPEG required on the client machine to work?

### ISSUE THREAD

**Issue by ollyde** [ekibun/flutter_ffplay#4]
As title

Or does it work standalone? For example if we distribute to a company?

**Comment by ekibun** [comment]
the compile process link the static library into a single library: flutter_ffmpeg.dll
client does not required extra ffmpeg program but the plugin generated DLL should be included

**Comment by ollyde** [comment]
@ekibun 
Alrighty; any reason why you can’t just include the complied version inside the library instead of us trying to compile it? (I can’t figure that out seems to be a lack of docs)

**Comment by ekibun** [comment]
Include the complied version for all platforms and archtectures will exceed the publish size limit.
My original consideration is to export all the compile script for users to configure their ffmpeg and reduce the dll size, but sorry for my poor English :(
To compile the ffmpeg, please use the git version and check the following discussion
https://github.com/ekibun/flutter_ffplay/issues/2#issuecomment-1109771129
Recently I have no time to refactor the build process. Sorry

**Comment by ollyde** [comment]
@ekibun thanks but you're comments don't work, don't know any who is using MVS2019 to build flutter apps? You've not explained what ```msys2`` and how to use it etc. Do we need to actually install Visual Studio (not code) to make this work? Which extensions of VS? 100 more questions.

I wasn't aware there was a limit? Wasn't that in 100's of GB?

Sure, could be nice if you had a link to a Google Drive or something else with the complied libraries and how to use them :-)

**Comment by ekibun** [comment]
First of all, this plugin ONLY support android and windows platforms.
`msys2` is an unix bash environment **for windows** that required to configure ffmpeg:
https://trac.ffmpeg.org/wiki/CompilationGuide/MSVC
If you do not use windows, just directly run `cxx/build.sh`.

The pub package is limited to be smaller than 100MB. ONLY one of the compiled libraries may exceed the size limit!

The compiled libraries depends on the system libraries and compiler version of your own environment. I have on time and have no idea to help you build the libraries.

**Comment by ollyde** [comment]
@ekibun yeah we only need it for Windows, seems that this is only viable video player plugin at the moment. The Dart VLC Kit plugin has so many bugs.

**Comment by vinayhvaghamshi** [comment]
I/flutter: Null check operator used on a null value
    #0      IOHandler._functions.<anonymous closure> (package:flutter_ffplay/src/io.dart:68:33)
